python/cudf/cudf/core/series.py

```diff
@@ -4035,26 +4007,106 @@ def describe_numeric(self):
             data = _format_stats_values(data)
 
             return Series(
-                data=data, index=names, nan_as_null=False, name=self.name,
+                data=data, index=index, nan_as_null=False, name=self.name,
+            )
+
+        def describe_timedelta(self):
+            # mimicking pandas
+            index = (
+                ["count", "mean", "std", "min"]
+                + _format_percentile_names(percentiles)
+                + ["max"]
+            )
+
+            data = (
+                [self.mean().to_numpy(), self.std().to_numpy(), self.min()]
+                + self.quantile(percentiles).to_array(fillna="pandas").tolist()
+                + [self.max()]
+            )
+            data = column.as_column(data, dtype=self.dtype)
+
+            data = [self.count()] + data.astype("str").to_array(
```

Comment:
Maybe I'm not following, but it looks like we're going from device to host in line 4021, then back from host to device in line 4026, and again device to host in line 4028. Can we eliminate one or more of these?